### PR TITLE
A stab at fixing #34: don't add CRLF to just any CDATA

### PR DESCRIPTION
**Pull request by wdoekes**
Not sure why the removed_crlf was only use in the `<send>` code and not the `<sendCmd>`.

Rob, do you see anything wrong with this?

### CODE REVIEW

**Comment by wdoekes** [comment]
@rkday and perhaps we should give a particular someone collaborator powers?

File: src/scenario.cpp [hunk 1]
```diff
@@ -647,6 +644,15 @@ char *clean_cdata(char *ptr, int *removed_crlf = NULL)
         memmove(ptr, ptr + 1, strlen(ptr));
     }
 
+    /* Re-add our removed (CR)LFs. */
+    while (removed_crlf--) {
+        strcat(msg, "\n");
+    }
+    /* A SIP message must contain a double (CR)LF to signify EOH. */
```
Comment (rkday) [inline src/scenario.cpp]: I think it must actually contain a double CRLF - the CR is mandatory. Worth a comment to say that the \r gets added by the SendingMessage constructor so we don't need it here.

**Comment by rkday** [comment]
Looks good, maybe worth a test with:

```
<send>
  <![CDATA[
      SIP/2.0 200 OK
      [last_Via:]
      [last_From:]
      [last_To:];tag=TAG
      [last_Call-ID:]
      [last_CSeq:]
      Contact: sip:sipp@127.0.0.1:5070
      Max-Forwards: 70]]>
  </send>
```

to ensure that EOH gets added even if it's not there.

File: src/scenario.cpp (hunk 1 above)
Comment (wdoekes) [inline src/scenario.cpp]: Yeah. It's mandatory in SIP. It was an attempt to document that the CR wasn't used here. But you're right: it's better to be explicit (ooh, zen of python).

**Comment by wdoekes** [comment]
Okay. Added the relevant checks.

Now I'm going nuts though, since it seems already fixed, even with an ancient (Ubuntu supplied sip-teste) sipp 1:3.2-1build1. I have absolutely no idea why I cannot reproduce the failing test anymore :-/

**Comment by wdoekes** [comment]
Okay. This changes the newly commited (6a09339) github-#0034 patch to fail:
```
Testing github-#0034: failed (on notify-manylf)
```
The cleanup is not worth the change in behaviour.

**Pull request closed without merging**